hack/update-codegen.sh

```diff
@@ -36,6 +39,39 @@ CURRENT_DIR=$(dirname $0)
 PROJECT_ROOT="${CURRENT_DIR}"/..
 export PROJECT_ROOT
 
+get_groups() {
+  local which=("$@")
+  local groups=()
+
+  for entry in "${which[@]}"; do
+    groups+=("${entry}_groups")
+  done
+
+  echo "${groups[@]}"
+}
+
+parse_flags() {
+  while test $# -gt 0; do
+    case "$1" in
+      --mode)
+        shift
+        if [[ -n "$1" ]]; then
+        MODE="$1"
+        fi
+        ;;
+      --which)
```

Comment:
Can we find a better name for the which flag? Initially, it was not obvious to us that it represents which groups to generate. We were wondering what is which - the directories, the packages or something else. Maybe can just call it `--groups` for simplicity?